pages/docs/modules/exec.mdx

```diff
@@ -0,0 +1,355 @@
+# exec
+
+## Functions
+
+### command
+
+```go filename="Function signature"
+command(file string, args ...string) exec.command
+```
+
+Creates a new `exec.command` targeting a specific `file` to execute,
+followed by arguments to execute the command with.
+
+Nothing is executed by this function alone, but instead only a [`exec.command`](#type-command)
+object is returned. To execute `file`, you must call the [`run`](#commandrun)
+method on the [`exec.command`](#type-command) object.
+
+The lookup if `file` executable exists is also not performed until the [`run`](#commandrun)
+method is called.
+
+```go copy filename="Example"
+>>> exec.command("bash")
+exec.command("bash")
+>>> exec.command("bash", "-c", "echo hello world")
+exec.command("bash", "-c", "echo hello world")
+>>> exec.command("file-that-does-not-exist")
+exec.command("file-that-does-not-exist")
+>>> cmd := exec.command("date", "-Iseconds")
+>>> cmd.run()
+>>> cmd.stdout
+"2024-01-09T21:33:48+01:00\n"
+```
+
+### look_path
+
+```go filename="Function signature"
+look_path(file string) string
+```
+
+Searches for an executable named `file` in the directories named by the `PATH`
+environment variable.
+If `file` contains a slash, it is tried directly and the `PATH` is not consulted.
+Otherwise, on success, the result is an absolute path.
+
+```go copy filename="Example"
+>>> exec.look_path("bash")
+"/usr/bin/bash"
+```
+
+### exec
+
+```go filename="Function signature"
+exec(file string, args list = [], params map = {}) exec.result
+```
+
+Searches for an executable named `file`, similarly to the `exec.look_path`
+function, and then executes it. The optional parameter `args` are passed as
+arguments when executing `file`, and the optional parameter `params` allows
+additional configurations for how to execute the `file`.
+
+In effect, it creates a new [`exec.command`](#type-command) object,
+configures it using the `params`,
+and then executes its [`run`](#commandrun) method.
+Except this function returns [`exec.result`](#type-result) instead of
+[`exec.command`](#type-command).
+
+```go copy filename="Example"
+>>> exec("date")
+exec.result(pid: 13289)
+>>> exec("date").stdout
+"Tue Jan  9 09:21:04 PM CET 2024\n"
+>>> exec("date", ["-Iseconds", "--utc"]).stdout
+"2024-01-09T20:22:43+00:00\n"
+>>> exec("date", ["-Iseconds"], {"env": {"TZ": "America/New_York"}}).stdout
+"2024-01-09T15:24:57-05:00\n"
+```
+
+Available configuration options for `params`:
+
+```json
+{
+  "stdout": os.stdout, // type: io.Writer
+  "stderr": os.stderr, // type: io.Writer
+  "stdin": os.stdin, // type: io.Reader | string | byte_slice
+  "dir": "/foo/bar", // working directory
+  "env": {
+    // values must be string
+    "MY_VAR": "123",
+    "TZ": "Europe/Berlin"
+  }
+}
+```
+
+## Type command
```

Comment:
I'm not 💯 about the header naming here. I don't want to name it just `command` as it would then collide with the slug/fragment ID for the function called `command`.

Could maybe make it more obvious with a delimiter, such as:

```suggestion
## Type: command
```

Wdyt?